recipes-openxt/xenclient/xenclient-tpm-scripts/tpm-functions

```diff
@@ -614,15 +615,15 @@ tpm2_write_tboot_policy() {
     fi
 
     tpm2_nvwrite -x "${tboot_idx}" -a "${TPM_RH_OWNER}" -P "${password}" \
-                 -f "${policy}" >/dev/null 2>&1
+                 "${policy}" >/dev/null 2>&1
 }
 
 tpm_get_pcr() {
     local pcr="${1}"
 
     is_tpm_2_0
     if [ $? -eq 0 ]; then
-        echo "$(tpm2_listpcrs|grep PCR_${pcr}|tail -1|cut -f2 -d:|tr 'A-Z' 'a-z'|tr -d \ )"
+        echo "$(tpm2_pcrlist -L sha256:${pcr} | tail -1 | cut -f2 -d: | tr 'A-Z' 'a-z' | tr -d \ )"
```

Comment:
Would it be acceptable to default to sha256 and use sha1 if not found?

Reply:
You could query the list of supported algorithms first with ```tpm2_pcrlist -s```, check if sha256 is there.  I would say its reasonable to default to sha256, and only use sha1 as a fallback since we know sha1 has collisions.

Reply:
Again, pardon the lacking clarity. I am good if the default is sha256, I was just pointing out that you hard coded sha256 into your parsing string. It would be better if something like ${alg} was used as it has been done elsewhere.

Reply:
I am going to do what @crogers1 suggested. That seems to be how we handle this for other operations.